packages/inferno/src/core/component.ts

```diff
@@ -74,7 +77,9 @@ export function rerender() {
 
   while ((component = QUEUE.shift())) {
     if (!component.$UN) {
-      applyState(component, false);
+      const force = component.$F;
+      component.$F = false;
```

Comment:
Why is this line needed?

Reply:
As `$F` stores the information whether changes should be forced or not, we assume that they are not forced by default unless `forceUpdate` is called. The default value of `$F` is false.

If `forceUpdate` is called we switch `$F` to true, and wait before it will be time for our component to render. After the component updates, we want to return the default value for the component. So, it's why I'm using `$F = false`. 

And finally, the component potentially could plan further changes via `setState` or even `forceUpdate` during its updating cycle. It means we want to set up all defaults before the component starts its updating cycle. So I cache value in `const force`.